### PR TITLE
add comments work in progress

### PR DESCRIPTION
**Pull request by MIYankov**
http://teampulse.telerik.com/view#item/345079

### CODE REVIEW

File: home/home-page.ts [hunk 1]
```diff
@@ -18,6 +24,6 @@ export function onNavigatingTo(args: EventData) {
 }
```
Comment (manoldonev) [inline home/home-page.ts]: "This gets a reference this page’s <StackLayout> UI component." -> "This gets a reference to this page’s <StackLayout> UI component."

File: home/home-page.xml [hunk 2]
```diff
@@ -1,29 +1,54 @@
+<!--
+The markup in NativeScript apps contains a series of user interface components, each
+of which NativeScript renders with a platform-specific iOS or Android native control.
+You can find a full list of user interface components you can use in your app at
+https://docs.nativescript.org/ui/components.
```
Comment (manoldonev) [inline home/home-page.xml]: Useful link here 👍

File: home/home-page.xml [hunk 3]
```diff
@@ -1,29 +1,54 @@
...
+-->
 <Page
```
Comment (manoldonev) [inline home/home-page.xml]: Should we describe the "Page" abstraction here?

File: home/home-page.xml [hunk 4]
```diff
@@ -1,29 +1,54 @@
...
             <nsDrawer:SlideInOnTopTransition/>
         </nsDrawer:RadSideDrawer.drawerTransition>
 
         <nsDrawer:RadSideDrawer.drawerContent>
-        <!--Add your drawer content here. If it's a custom component, add the xmlns to the DrawerPage tag above-->
-        <StackLayout>
-            <myDrawer:MyDrawer />
-        </StackLayout>
-        <!--End of drawer content-->
+            <!-- The RadSideDrawer content should always be wrapped in a layout component -->
```
Comment (manoldonev) [inline home/home-page.xml]: Such gotchas would be very useful 👍

File: home/home-view-model.ts [hunk 5]
```diff
@@ -1,5 +1,16 @@
...
+* data to the view. NativeScript provides a modules called 'Observable',
+* which facilitates creating a view model object that can be bound to the view.
+* Learn more about the basic application logic in this documentation article:
+* https://docs.nativescript.org/tutorial/chapter-3#chapter-3application-logic
+*************************************************************/
+
 import { Observable } from "data/observable";
 
+/* ***********************************************************
+* The HomeViewModel class is referenced in the code-behind file home-page.ts
```
Comment (manoldonev) [inline home/home-view-model.ts]: Do we have an idea where / how to describe the relationship between the view, code behind file and the view model?

File: shared/my-drawer/MyDrawer-view-model.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                route: "featured/featured-page"
+            },
+            {
+                title: "Settings",
+                route: "settings/settings-page"
+            }
+        ];
+    }
+
+    get navigationItems(): Array<any> {
```
Comment (manoldonev) [inline shared/my-drawer/MyDrawer-view-model.ts]: Should we explain getters / setters as well?

File: shared/my-drawer/MyDrawer.ts [hunk 7]
```diff
@@ -1,48 +1,68 @@
...
-            title: "Settings",
-            route: "settings/settings-page"
-        }
-    ]
-};
+import { MyDrawerViewModel } from "./MyDrawer-view-model";
 
-// Event handler for Page "navigatingTo" event attached in tabs-page.xml
+/* ***********************************************************
+* Custom components are not Pages and they don't have the navigation event handlers of a Page.
```
Comment (manoldonev) [inline shared/my-drawer/MyDrawer.ts]: Useful 👍

File: home/home-page.ts (hunk 1 above)
Comment (MIYankov) [inline home/home-page.ts]: There has been major changes to the comments.

File: shared/my-drawer/MyDrawer-view-model.ts (hunk 6 above)
Comment (MIYankov) [inline shared/my-drawer/MyDrawer-view-model.ts]: There has been major changes to the comments.

**Pull request merged**